src/Translation_unit.ml

```diff
@@ -406,18 +279,138 @@ let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
                       Location.print_loc loc (ellipsis_cmt s) )
           | `Comment_dropped l when not conf.Conf.quiet ->
               List.iter l ~f:(fun (loc, msg) ->
-                  Caml.Format.eprintf
+                  Caml.Format.fprintf fmt
                     "%!@{<loc>%a@}:@,@{<error>Error@}: Comment (* %s *) \
                      dropped.\n\
                      %!"
                     Location.print_loc loc (ellipsis_cmt msg) )
           | _ -> () ) ;
           if Conf.debug then
             List.iter l ~f:(fun (msg, sexp) ->
-                Format.eprintf "  %s: %s\n%!" msg (Sexp.to_string sexp) )
+                Caml.Format.fprintf fmt "  %s: %s\n%!" msg
+                  (Sexp.to_string sexp) )
       | exn ->
-          Format.eprintf
+          Caml.Format.fprintf fmt
             "  BUG: unhandled exception. Use [--debug] for details.\n%!" ;
-          if Conf.debug then Format.eprintf "%s\n%!" (Exn.to_string exn) )
-  ) ;
+          if Conf.debug then
+            Caml.Format.fprintf fmt "%s\n%!" (Exn.to_string exn) )
+
+let format xunit (conf : Conf.t) ?output_file ~input_name ~source ~parsed ()
+    =
+  let dump_ast ~suffix ast =
+    dump_ast ~input_name ?output_file ~suffix (fun fmt ->
+        xunit.printast fmt ast )
+  in
+  let dump_formatted = dump_formatted ~input_name ?output_file in
+  Location.input_name := input_name ;
+  (* iterate until formatting stabilizes *)
+  let rec print_check ~i ~(conf : Conf.t) (t : _ with_comments) ~source :
+      (string, error) Result.t =
+    dump_ast ~suffix:".old" t.ast ;
+    let format ~box_debug =
+      let buffer = Buffer.create (String.length source) in
+      let source_t = Source.create source in
+      let cmts_t = xunit.init_cmts source_t conf t.ast t.comments in
+      let fs = Format.formatter_of_buffer buffer in
```

Comment:
maybe change this Format to Format_, and the same for the next 2, and then remove the module alias at the top and `Caml.` prefixes on the other occurrences of Format?

Reply:
see commit I pushed to your branch